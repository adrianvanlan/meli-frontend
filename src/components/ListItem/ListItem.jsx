import IconShipping from '../IconShipping/IconShipping';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { getCurrencyPrice } from '../../helpers';
import styles from './ListItem.module.scss';

const listItem = ({ itemData }) => {
  const { price } = itemData;

  return (
    <li className={styles.ListItem} data-item-id={itemData && itemData.id}>
      <>
        <div className={styles.ListItem__Image}>
          <img src={itemData.picture} alt={`Imagen ${itemData.title}`} />
        </div>
        <div className={styles.ListItem__Info}>
          {itemData.price && (
            <div className={styles.ListItem__Info__Price}>
              {itemData.price.amount > 0 ? (
                <span>{getCurrencyPrice(price.amount, price.currency)}</span>
              ) : null}
              {itemData.price.decimals > 0 ? (
                <span className={styles.ListItem__Info__Price__Decimals}>
                  {itemData.price.decimals}
                </span>
              ) : null}
              {itemData.free_shipping && (
                <span className={styles.ListItem__Info__Price__Shipping}>
                  <IconShipping />
                </span>
              )}
            </div>
          )}
          {itemData.title && (
            <Link
              to={`/items/${itemData.id}`}
              className={styles.ListItem__Info__Link}
            >
              <h2 className={styles.ListItem__Info__Title}>{itemData.title}</h2>
            </Link>
          )}
        </div>
      </>
    </li>
  );
};

listItem.propTypes = {
  itemData: PropTypes.object.isRequired,
};

export default listItem;
