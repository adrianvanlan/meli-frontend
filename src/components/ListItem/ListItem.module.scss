@import '../../styles/base';

.ListItem {
  border-bottom: thin solid #eee;
  position: relative;
  width: 100%;
  background-color: $white;
  border-radius: 4px 4px 0 0;
  display: flex;
  flex-direction: row;
  padding: 20px 50px 20px 0;

  &__Image {
    border-radius: 4px;
    box-sizing: content-box;
    flex-shrink: 0;
    font-size: 1px;
    height: 160px;
    margin: 0;
    overflow: hidden;
    padding: 0 24px;
    width: 160px;

    img {
      width: 100%;
      height: 100%;
      object-fit: contain;
    }
  }

  &__Info {
    display: flex;
    flex-direction: column;
    width: calc(100% - 208px);
    align-items: flex-start;
    padding-top: 20px;
    max-width: 600px;
    text-align: left;

    &__Link {
      color: $primary-text;
      text-decoration: none;

      &::before {
        content: '';
        height: 100%;
        left: 0;
        position: absolute;
        top: 0;
        width: 100%;
        z-index: 0;
      }
    }

    &__Title {
      color: #333;
      font-size: 20px;
      font-weight: 300;
      line-height: 1.3;
    }

    &__Price {
      color: $primary-text;
      display: flex;
      font-size: 24px;
      font-weight: 400;
      line-height: 1.25;
      align-items: center;

      span {
        line-height: 1;
      }

      &__Decimals {
        align-self: flex-start;
        font-size: 14px;
        font-weight: 400;
        margin: 2px 0 0 2px;
      }

      &__Shipping {
        color: $primary-link;
        font-size: 14px;
        font-weight: 400;
        line-height: 1.25;
        margin-left: 10px;
      }
    }
  }
}
