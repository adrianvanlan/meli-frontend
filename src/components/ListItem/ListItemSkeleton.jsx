import Skeleton from 'react-loading-skeleton';
import styles from './ListItem.module.scss';

const ListItemSkeleton = () => {
  return (
    <li className={styles.ListItem}>
      <div className={styles.ListItem__Image}>
        <Skeleton width={160} height={160} />
      </div>
      <div className={styles.ListItem__Info}>
        <div className={styles.ListItem__Info__Price}>
          <Skeleton height={20} width={150} />
        </div>
        <div className={styles.ListItem__Info__Link}>
          <div className={styles.ListItem__Info__Title}>
            <Skeleton height={20} width={600} style={{ marginTop: '20px' }} />
          </div>
        </div>
      </div>
    </li>
  );
};

export default ListItemSkeleton;
