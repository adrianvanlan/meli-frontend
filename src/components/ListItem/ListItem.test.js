import ListItem from './ListItem';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

it('expect to render ListItem component', () => {
  const mockItemData = {
    id: 'MLAEXAMPLE',
    title: 'some title',
  };
  const wrapper = render(
    <MemoryRouter>
      <ListItem itemData={mockItemData} />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});
