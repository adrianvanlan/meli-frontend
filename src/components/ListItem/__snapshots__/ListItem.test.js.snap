// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ListItem component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="ListItem"
        data-item-id="MLAEXAMPLE"
      >
        <div
          class="ListItem__Image"
        >
          <img
            alt="Imagen some title"
          />
        </div>
        <div
          class="ListItem__Info"
        >
          <a
            class="ListItem__Info__Link"
            href="/items/MLAEXAMPLE"
          >
            <h2
              class="ListItem__Info__Title"
            >
              some title
            </h2>
          </a>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="ListItem"
      data-item-id="MLAEXAMPLE"
    >
      <div
        class="ListItem__Image"
      >
        <img
          alt="Imagen some title"
        />
      </div>
      <div
        class="ListItem__Info"
      >
        <a
          class="ListItem__Info__Link"
          href="/items/MLAEXAMPLE"
        >
          <h2
            class="ListItem__Info__Title"
          >
            some title
          </h2>
        </a>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
